### PR TITLE
Added GetDevice Logic

### PR DESCRIPTION
**Pull request by Lyddon7522**
Feels like the nulls are gross but 🤷 . I've got to add tests still

### CODE REVIEW

File: src/MedTrackerAPI/Features/Devices/GetDevice.cs [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    }
+    
+    public class GetDeviceQuery : IRequest<Response>
+    {
+        public int DeviceId { get; set; }
+    }
+    
+    public class GetDeviceQueryHandler(MedTrackerDbContext context) : IRequestHandler<GetDeviceQuery, Response>
+    {
+        public async Task<Response> Handle(GetDeviceQuery request, CancellationToken cancellationToken)
```
Comment (benjaminsampica) [inline src/MedTrackerAPI/Features/Devices/GetDevice.cs]: You can return `Response?` so that the compiler can give warnings to callers and you don't need the null forgiving (!) operator - aka damnit operator - later on.

File: src/MedTrackerAPI/Features/Devices/GetDevice.cs [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+        { 
+            var device = await context.Devices.Where(d => d.Id == request.DeviceId).SingleOrDefaultAsync(cancellationToken);
+   
+            if (device is null)
```
Comment (benjaminsampica) [inline src/MedTrackerAPI/Features/Devices/GetDevice.cs]: Code golf ->

`return device is null ? null : new Response(device.Id, device.Description, device.Manufacturer, device.Model, device.PartNumber, device.LotNumber);`

Par one

`return (await context.Devices.SingleOrDefaultAsync(d => d.Id == request.DeviceId, cancellationToken)) is { } device ? new Response(device.Id, device.Description, device.Manufacturer, device.Model, device.PartNumber, device.LotNumber) : null;`

File: src/MedTrackerAPI/Features/Devices/GetDevice.cs [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    public class GetDeviceQuery : IRequest<Response>
+    {
+        public int DeviceId { get; set; }
+    }
+    
+    public class GetDeviceQueryHandler(MedTrackerDbContext context) : IRequestHandler<GetDeviceQuery, Response>
+    {
+        public async Task<Response> Handle(GetDeviceQuery request, CancellationToken cancellationToken)
+        { 
+            var device = await context.Devices.Where(d => d.Id == request.DeviceId).SingleOrDefaultAsync(cancellationToken);
```
Comment (benjaminsampica) [inline src/MedTrackerAPI/Features/Devices/GetDevice.cs]: If you're doing a PK lookup, you can use `FindAsync()`. Passing the cancellationToken is kind of wonky to this method, but here's what you're looking for. 

`var device = await context.Devices.FindAsync([id], cancellationToken)`

If you're wanting to stick with `SingleOrDefaultAsync`, you can elide the `.Where()` and pass it to `.SingleOrDefaultAsync()` like `.SingleOrDefaultAsync(d => d.Id == request.Id, cancellationToken)`

If you're willing to bring in a library, AutoMapper is a good package for doing projection for queries that are simple like this. It has some benefits for testing, too. I'm being reductive, but here's an example of how AutoMapper + EF would do it

```
return context.Devices
    .ProjectTo<Response>(mapper.ConfigurationProvider)
    .SingleOrDefaultAsync(d => d.Id == request.Id, cancellationToken); // returns null if not found still.
```

File: src/MedTrackerAPI/Features/Devices/GetDevice.cs [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+{
+    public record Response(int DeviceId, string Description, string Manufacturer, string Model, string PartNumber, string LotNumber);
+    
+    public sealed class Endpoint : IEndpoint
+    {
+        public void MapEndpoint(IEndpointRouteBuilder app)
+        {
+            app.MapGet("devices/{deviceId}", async (IMediator mediator, int deviceId) =>
+                {
+                    var query = new GetDeviceQuery { DeviceId = deviceId };
```
Comment (benjaminsampica) [inline src/MedTrackerAPI/Features/Devices/GetDevice.cs]: I normally scrunch simple lines like this together w/ the next but that's my preference.

File: src/MedTrackerAPI/Features/Devices/GetDevice.cs [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+                    var device = await mediator.Send(query);
+                    
+                    return device is null ? Results.BadRequest("Device not found") : Results.Ok(device);
```
Comment (benjaminsampica) [inline src/MedTrackerAPI/Features/Devices/GetDevice.cs]: Um ackshually, if you're wanting to follow the HTTP spec this would be a 404 (Not Found), not a 400 (Bad Request). 404's don't usually have a body to them, with the 404 being enough to convey "Not found"

File: src/MedTrackerAPI/Features/Devices/GetDevice.cs (hunk 3 above)
Comment (Lyddon7522) [reply src/MedTrackerAPI/Features/Devices/GetDevice.cs]: I didn't put in AutoMapper but that's mostly because I don't fully understand how to or why to implement it. I might have you ELI5 sometime with that one.

**Review by benjaminsampica — APPROVED** [review]
